sql/parser/eval_test.go

```diff
@@ -308,3 +315,25 @@ func TestEvalExprError(t *testing.T) {
 		}
 	}
 }
+
+func TestSimilarEscape(t *testing.T) {
+	testData := []struct {
+		expr     string
+		expected string
+	}{
+		{`test`, `test`},
+		{`test%`, `test.*`},
+		{`_test_`, `.test.`},
+		{`_%*`, `..**`},
+		{`[_%]*`, `[_%]*`},
+		{`.^`, `\.\^`},
+		{`%(b|d)%`, `.*(?:b|d).*`},
+		{`se\"arch\"[\"]`, `se(arch)[\"]`},
```

Comment:
Yeah, I saw those docs after I sent my comment. Very odd syntax. It isn't clear to me if that escaped double-quote transforms is necessary for all regexes or just SUBSTRING ones.